src/Hl7.Fhir.Specification/Specification/Source/DirectorySourceSettings.cs

```diff
@@ -0,0 +1,273 @@
+        /// </value>
+        /// <example>
+        /// <code>Includes = new string[] { "profiles/**/*", "**/valuesets" };</code>
+        /// </example>
+        public string[] Includes { get; set; }
+
+        /// <summary>
+        /// Gets or sets an array of search strings to match against the names of subdirectories of the content directory.
+        /// The source will ignore all subdirectories that match the specified exclude mask(s).
+        /// The source will only provide resources from subdirectories that don't match the specified exclude mask(s).
+        /// </summary>
+        /// <remarks>
+        /// Exclude filters are applied last, after any <see cref="Mask"/> and <see cref="Includes"/> filters.
+        /// </remarks>
+        /// <value>
+        /// Supported wildcards:
+        /// <list type="bullet">
+        /// <item>
+        /// <term>*</term>
+        /// <description>Matches zero or more characters within a directory name.</description>
+        /// </item>
+        /// <item>
+        /// <term>**</term>
+        /// <description>
+        /// Recursive wildcard.
+        /// For example, <c>/hello/**/*</c> matches all descendants of <c>/hello</c>.
+        /// </description>
+        /// </item>
+        /// </list>
+        /// </value>
+        /// <example>
+        /// <code>Excludes = new string[] { "profiles/**/old", "temp/**/*" };</code>
+        /// </example>
+        public string[] Excludes { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value that determines how to process duplicate files with multiple serialization formats.
+        /// <para>
+        /// Returns <see cref="DirectorySource.DuplicateFilenameResolution.PreferXml"/> by default.
+        /// </para>
+        /// </summary>
+        public DirectorySource.DuplicateFilenameResolution FormatPreference { get; set; } = DirectorySource.DuplicateFilenameResolution.PreferXml;
+
+        /// <summary>Gets or sets a custom <see cref="NavigatorStreamFactory"/> delegate.</summary>
+        /// <remarks>
+        /// The <see cref="ArtifactSummaryGenerator"/> depends on the <see cref="INavigatorStream"/>
+        /// interface to extract summary information from FHIR artifacts, independent of the underlying
+        /// resource serialization format. By default, the <see cref="DirectorySource"/> calls the
+        /// <see cref="DefaultNavigatorStreamFactory.Create(string)"/> method to create
+        /// a concrete navigator stream for the current file. The default factory supports
+        /// navigators for "*.xml" and "*.json" files.
+        /// The caller can override the default navigator streams by specifying a custom
+        /// <see cref="NavigatorStreamFactory"/> delegate. This allows clients to implement
+        /// support for alternative serialization formats.
+        /// </remarks>
+        public NavigatorStreamFactory StreamFactory { get; set; } // = DefaultNavigatorStreamFactory.Create;
```

Comment:
To align with the rest of the API, I think I prefer an event-based approach here, instead of a factory.

Settings.OnNavigatorStreamNeeded, with the given path as a parameter to the event

Reply:
Sure! And also rewrite SummaryFactory delegate to event?

Reply:
I now understand you want to do this, not so much to support customization of creation of the navigator stream, but to have an extensible way to support more (future) formats of FHIR.  This is a good idea, but we should generalize that - i.e. how would I know which parser to create for a given format?  (not just the stream).  Let's think about that for a bit before deciding on this.

Reply:
I'd like to know what @christiaan and @martijn think about this too.